### ISSUE TITLE
BaseGameActivity crashed!

### ISSUE THREAD

**Issue by muhanov** [nicolasgramlich/AndEngine#167]
Hi,
Somethimes BaseGameActivity is crashed after power off/on button is clicked.
See a stacktrace:
08-27 23:22:51.160: E/AndroidRuntime(565): FATAL EXCEPTION: main
08-27 23:22:51.160: E/AndroidRuntime(565): java.lang.NullPointerException
08-27 23:22:51.160: E/AndroidRuntime(565):  at org.andengine.ui.activity.BaseGameActivity.onResumeGame(BaseGameActivity.java:222)
08-27 23:22:51.160: E/AndroidRuntime(565):  at org.andengine.ui.activity.BaseGameActivity$4.run(BaseGameActivity.java:373)
08-27 23:22:51.160: E/AndroidRuntime(565):  at android.os.Handler.handleCallback(Handler.java:615)
08-27 23:22:51.160: E/AndroidRuntime(565):  at android.os.Handler.dispatchMessage(Handler.java:92)
08-27 23:22:51.160: E/AndroidRuntime(565):  at android.os.Looper.loop(Looper.java:137)
08-27 23:22:51.160: E/AndroidRuntime(565):  at android.app.ActivityThread.main(ActivityThread.java:4745)
08-27 23:22:51.160: E/AndroidRuntime(565):  at java.lang.reflect.Method.invokeNative(Native Method)
08-27 23:22:51.160: E/AndroidRuntime(565):  at java.lang.reflect.Method.invoke(Method.java:511)
08-27 23:22:51.160: E/AndroidRuntime(565):  at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:786)
08-27 23:22:51.160: E/AndroidRuntime(565):  at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:553)
08-27 23:22:51.160: E/AndroidRuntime(565):  at dalvik.system.NativeStart.main(Native Method)

This is because a possible race condition here.
Actually, onResumeGame can be called from a separate thread in function callGameResumedOnUIThread.
But to this time onDestroy can be already called for this activity and mEngine is null and new copy of Activity is created separately.
Main game thread is interrupted in Engine.run() but this thread is not interrupted.

I propose to change callGameResumedOnUIThread as follow:
    private void callGameResumedOnUIThread() {
        BaseGameActivity.this.runOnUiThread(new Runnable() {
            @Override
            public void run() {
                if (isGameLoaded()) {
                    BaseGameActivity.this.onResumeGame();
                }
            }
        });
In other words, we will call onResume only in case when game is loaded.
But if this thread is invoked after onDestroy, this call is skipped.

Maxim
